langchain/evaluation/run_evaluators/string_run_evaluator.py

```diff
@@ -331,7 +341,7 @@ def from_model_and_evaluator(
                 " not yet implemented."
                 "Expected one of [BaseLanguageModel, Chain, Tool]."
             )
-        if reference_key is not None:
+        if reference_key is not None or isinstance(model, BaseLanguageModel):
```

Comment:
Hm `reference_key` here was also communicating whether an example was required at all. May be tricky

Reply:
That does not seem to be how reference_key is used inside StringExampleMapper, see

<img width="751" alt="Screenshot 2023-07-07 at 21 09 53" src="https://github.com/hwchase17/langchain/assets/56902/47f20324-ed84-41d6-a4c8-4b93bdea85c6">

You always pick a reference even if reference_key is None

Reply:
Here is a possible diff that maintains error message while not requiring reference_key for llm/chat examples

<img width="1215" alt="Screenshot 2023-07-07 at 21 13 40" src="https://github.com/hwchase17/langchain/assets/56902/df7dccd5-8d43-4a59-a21c-1c802d96f99c">

Reply:
I like that. Will make the change.

Indeed within the example mapper it was fine, I was just raising an error in the run evaluator